src/cabin/cabin.service.ts

```diff
@@ -1,14 +1,115 @@
 import { Injectable } from '@nestjs/common';
+import { VisbookApi } from 'src/visbook/visbook.api';
 import { CabinDetails, CabinApi } from './cabin.api';
+import { dateIsValid, getVisbookId } from './cabin.utils';
 
 @Injectable()
 export class CabinService {
   private readonly apiClient = new CabinApi();
+  private readonly visbookApiClient = new VisbookApi();
+
   async getRandomCabin(): Promise<CabinDetails> {
     const cabins = await this.apiClient.getCabins();
     const cabin = cabins[Math.floor(Math.random() * cabins.length)];
     const { id } = cabin.node;
     const cabinDetails = await this.apiClient.getCabinDetails(id);
     return cabinDetails;
   }
+
+  async isCabinAvailable(
+    cabin: CabinDetails,
+    checkIn: string,
+    checkOut: string,
```

Comment:
I think it would be better if these were Date objects rather than strings. This also means that the validation gets pushed to where the user input is received which is a good thing IMO.

Reply:
https://github.com/dnt-frivillig/dnt-discord-bot/issues/12
Created an issue for this to be implemented in a future version of this feature.